workers/repomirrorworker/__init__.py

```diff
@@ -257,14 +259,33 @@ def perform_mirror(skopeo, mirror):
         return
     finally:
         if overall_status == RepoMirrorStatus.FAIL:
+            log_tags = []
+            log_message = "'%s' with tag pattern '%s'"
+
+            # Handle the case where not all tags were synced and state will not be rolled back
+            if (
+                len(failed_tags) != len(tags)
```

Comment:
Could this be simplified by just checking if there's `failed_tags`? Is there a situation where `failed_tags` and `tags` would be equal?

Reply:
If `failed_tags` is equal to `tags` it isn't considered a partial failure, since all tags have failed instead of only some.